### ISSUE TITLE
Crash report - Index 1 is out of bound

### ISSUE THREAD

**Issue by jcreedcmu** [github/hub#856]
Error (*errors.errorString): `Index 1 is out of bound`

Stack:

```
goroutine 1 [running]:
github.com/github/hub/commands.(*Args).InsertParam(0xc20805e1b0, 0x1, 0xc2082b9d38, 0x4, 0x4)
	/var/folders/w8/dp0lmdjs2vs51lsxbrf4s49h0000gn/T/go/src/github.com/github/hub/commands/args.go:99 +0x170
github.com/github/hub/commands.transformCheckoutArgs(0xc20805e1b0, 0x0, 0x0)
	/var/folders/w8/dp0lmdjs2vs51lsxbrf4s49h0000gn/T/go/src/github.com/github/hub/commands/checkout.go:102 +0xd63
github.com/github/hub/commands.checkout(0x6876a0, 0xc20805e1b0)
	/var/folders/w8/dp0lmdjs2vs51lsxbrf4s49h0000gn/T/go/src/github.com/github/hub/commands/checkout.go:36 +0x40
github.com/github/hub/commands.(*Command).Call(0x6876a0, 0xc20805e1b0, 0x0, 0x0)
	/var/folders/w8/dp0lmdjs2vs51lsxbrf4s49h0000gn/T/go/src/github.com/github/hub/commands/commands.go:46 +0x189
github.com/github/hub/commands.(*Runner).Call(0xc20802b120, 0x6876a0, 0xc20805e1b0, 0x0, 0x0, 0x0)
	/var/folders/w8/dp0lmdjs2vs51lsxbrf4s49h0000gn/T/go/src/github.com/github/hub/commands/runner.go:89 +0x54
github.com/github/hub/commands.(*Runner).Execute(0xc20802b120, 0x0, 0x0, 0x0)
	/var/folders/w8/dp0lmdjs2vs51lsxbrf4s49h0000gn/T/go/src/github.com/github/hub/commands/runner.go:81 +0x215
main.main()
	/private/tmp/hub-xGkPdT/hub-2.2.0/main.go:12 +0x40

```

Runtime:

```
GOOS: darwin
GOARCH: amd64
```

**Comment by jcreedcmu** [comment]
I was spuriously using the -p flag

**Comment by jcreedcmu** [comment]
in a commandline like

    git checkout -p git://github.com/$USER/$REPO/pull/1234 $BRANCH

**Comment by mislav** [comment]
Whenever a crash happens, it's a bug that we want to avoid, event if it was caused by an invalid flag. Does the crash happen for you every time with the latest version of hub? What is your `hub --version`? Thanks

**Comment by jcreedcmu** [comment]
```
$ hub --version
git version 2.3.3
hub version 2.2.0
```

**Comment by jcreedcmu** [comment]
And he crash happens consistently every time I run the above commandline, yes.

**Comment by jingweno** [comment]
I couldn't reproduce this with the `git-checkout -p`:

```
$ git checkout -p git://github.com/github/hub/pull/857 foo
Updating cesanta
From git://github.com/cesanta/hub
 * [new branch]      pull_assign -> cesanta/pull_assign
fatal: '--track' cannot be used with updating paths
```

I'm getting a fatal error instead of a crash report.

**Comment by jingweno** [comment]
I made some improvement in https://github.com/github/hub/pull/861 to help debug issues like this - I added the running command that causes the runtime exception and hub version.

**Comment by jingweno** [comment]
Okay, this bug was actually fixed in https://github.com/github/hub/commit/2884da15b2a73097d5b51dc40aa047e1ef2ec56b. I'm running the latest master locally. No wonder I couldn't reproduce it...

Similar bug: https://github.com/github/hub/issues/839.

@mislav How about releasing 2.2.1 sooner?

**Comment by mislav** [comment]
Sure, feel free to release any bugfixes from the 2.2-stable branch :+1:

**Comment by jingweno** [comment]
This bug will be fixed in v2.2.1. Closing.
